### PR TITLE
feat(Portable): Migrate more of DevSupport

### PR DESCRIPTION
**Pull request by kevinvangelder**
Still a few left, but we won't be bringing ShakeAccelerometer. The XAML ports are approximate and will likely need tweaking when we get a playground up and running (for example, none of the dialogs have cancel buttons).

### CODE REVIEW

File: ReactWindows/ReactNative.Net46/app.config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/app.config]: Is this file needed?

File: ReactWindows/ReactNative.Net46/packages.config [hunk 2]
```diff
@@ -1,14 +1,15 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Facebook.CSSLayout" version="1.1.1" targetFramework="net46" />
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/packages.config]: >Facebook.CSSLayout [](start = 15, length = 18)

We may have talked about this before, but we're using version 2.0.1-beta right now for Facebook.CSSLayout.

File: ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj [hunk 3]
```diff
@@ -61,8 +61,14 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="PresentationCore" />
+    <Reference Include="PresentationFramework" />
+    <Reference Include="SocketMessaging, Version=1.1.6002.26530, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\SocketMessaging.1.1.6002.26530\lib\net451\SocketMessaging.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj]: Nitpick, but the .NET Framework dependencies (i.e., those that don't come from NuGet or private drops, or those without HintPaths) typically come above the rest of the dependencies here.

File: ReactWindows/ReactNative.Net46/DevSupport/ProgressDialog.xaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            <RowDefinition Height="Auto" />
+            <RowDefinition />
+        </Grid.RowDefinitions>
+        <TextBlock Name="MessageTextBlock"
+                   Text="{Binding Path=Message, Mode=OneTime}"
+                   HorizontalAlignment="Center"
+                   Margin="0 10 0 0"
+                   Grid.Row="0" />
+        <ProgressBar IsEnabled="True"
+                      IsIndeterminate="True"
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/DevSupport/ProgressDialog.xaml]: >IsIndeterminate [](start = 22, length = 15)

Attributes have aligned whitespace in TextBlock above but not in ProgressBar here.

File: ReactWindows/ReactNative.Net46/Properties/Settings.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+namespace ReactNative.Net46.Properties {
+    
+    
+    [global::System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "14.0.0.0")]
+    internal sealed partial class Settings : global::System.Configuration.ApplicationSettingsBase {
+        
+        private static Settings defaultInstance = ((Settings)(global::System.Configuration.ApplicationSettingsBase.Synchronized(new Settings())));
+        
+        public static Settings Default {
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Properties/Settings.Designer.cs]: >Settings [](start = 22, length = 8)

What is this Settings designer for?

File: ReactWindows/ReactNative.Shared/DevSupport/DevServerHelper.cs [hunk 6]
```diff
@@ -7,8 +7,10 @@
 using System.Reactive.Disposables;
 using System.Threading;
 using System.Threading.Tasks;
+#if WINDOWS_UWP
 using Windows.System.Threading;
 using Windows.UI.Core;
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevServerHelper.cs]: Are both of these namespaces needed for UWP now that you changed from ThreadPool.Run -> Task.Run?

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 7]
```diff
@@ -1,6 +1,10 @@
 ﻿using ReactNative.Modules.DevSupport;
 using System.Collections.Generic;
+#if WINDOWS_UWP
 using Windows.Storage;
+#else
+using PCLStorage;
+#endif
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: Not seeing where this namespace is being used (since you're using the Settings.settings to read/write values)

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 8]
```diff
@@ -130,14 +135,25 @@ private T GetSetting<T>(string key, T defaultValue)
...
         }
 
         private void SetSetting<T>(string key, T value)
         {
+#if WINDOWS_UWP
             var values = ApplicationData.Current.LocalSettings.Values;
             values[key] = value;
+#else
+            ReactNative.Net46.Properties.Settings.Default[key] = value;
+#endif
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: I'm assuming you've tested this in some way?  Does the generated Settings class actually support arbitrary keys without having added anything to the designer?

Comment (kevinvangelder) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: Hmm, looks like probably not. Any suggestions?

File: ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj (hunk 3 above)
Comment (kevinvangelder) [inline ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj]: This came from NuGet.

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 9]
```diff
@@ -130,14 +136,37 @@ private T GetSetting<T>(string key, T defaultValue)
                     return (T)data;
                 }
             }
+#else
+            try
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: >try [](start = 12, length = 3)

I'd add some safety arround this...
```c#
if (typeof(T) == typeof(bool))
{
    ...
}
else
{
   throw new NotSupportedException($"Configuration values of type '{typeof(T)}' are not supported.");
}
```

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 10]
```diff
@@ -337,13 +363,32 @@ public Task<bool> IsPackagerRunningAsync()
                 await ReloadJavaScriptFromFileAsync(progressDialog.Token);
                 dialogOperation.Cancel();
             }
+#else
+            var dialogOperation = progressDialog.ShowDialog();
+
+            if (IsRemoteDebuggingEnabled)
+            {
+                await ReloadJavaScriptInProxyMode(progressDialog.Hide, progressDialog.Token).ConfigureAwait(false);
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: >progressDialog.Hide [](start = 50, length = 19)

Can you lift `progressDialog.Hide` and `dialogOperation.Cancel` into an variable of type Action so the bulk of this can be shared?

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 11]
```diff
@@ -533,14 +627,22 @@ public DevOptionHandler(string name, Action onSelect)
 
             public string Name { get; }
 
+#if WINDOWS_UWP
             public IAsyncInfo AsyncInfo { get; set; }
+#else
+            public Action AsyncInfo { get; set; }
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: >AsyncInfo [](start = 26, length = 9)

No need to keep the name the same if you make the change below (I don't see any other shared code that references the AsyncInfo property)

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 12]
```diff
@@ -533,14 +627,22 @@ public DevOptionHandler(string name, Action onSelect)
...
+#endif
 
             public void OnSelect()
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: >OnSelect [](start = 24, length = 8)

I would change the first part of this function to...
```c#
#if WINDOWS_UWP
                 AsyncInfo?.Cancel();
#else
                 AsyncInfo?.Invoke();
#endif
```

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 13]
```diff
@@ -533,14 +627,22 @@ public DevOptionHandler(string name, Action onSelect)
 
             public string Name { get; }
 
+#if WINDOWS_UWP
             public IAsyncInfo AsyncInfo { get; set; }
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: >IAsyncInfo [](start = 19, length = 10)

You don't have to keep IAsyncInfo here, you can consolidate these both into a single "hide" action, and pass in asyncInfo.Cancel as the property value for UWP directly.

**Comment by rozele** [comment]
:clock1:

File: ReactWindows/ReactNative.Net46/DevSupport/WebSocketJavaScriptExecutor.cs [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ReactNative.DevSupport
+{
+    class WebSocketJavaScriptExecutor : IJavaScriptExecutor
```
Comment (matthargett) [inline ReactWindows/ReactNative.Net46/DevSupport/WebSocketJavaScriptExecutor.cs]: Any tests, even minor ones, that could be added for this? What was the manual test case?

Comment (kevinvangelder) [inline ReactWindows/ReactNative.Net46/DevSupport/WebSocketJavaScriptExecutor.cs]: Unfortunately, no. I don't have any manual test case other than the fact that the interface is public unchanged and it compiles. :(

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 15]
```diff
@@ -130,14 +136,44 @@ private T GetSetting<T>(string key, T defaultValue)
...
+                    return (T) (object) result;
+                }
+                catch (ConfigurationErrorsException)
+                {
+
+                }
+            }
+            else
+            {
+                throw new NotSupportedException($"Configuration values of type '{typeof(T)}' are not supported.");
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: Use Invariant(...) here.  Sorry, didn't include it in my comment.

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 16]
```diff
@@ -130,14 +136,44 @@ private T GetSetting<T>(string key, T defaultValue)
...
+            {
+                try
+                {
+                    Boolean result = Boolean.TryParse(ConfigurationManager.AppSettings[key], out result);
+                    return (T) (object) result;
+                }
+                catch (ConfigurationErrorsException)
+                {
+
+                }
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: Is there nothing that needs to be done if there's a ConfigurationError?  Shouldn't you just let this exception through?

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs [hunk 17]
```diff
@@ -322,28 +347,35 @@ public Task<bool> IsPackagerRunningAsync()
                 : "Connecting to remote debugger.";
 
             var progressDialog = new ProgressDialog("Please wait...", message);
+#if WINDOWS_UWP
             var dialogOperation = progressDialog.ShowAsync();
-
+            Action cancel = dialogOperation.Cancel;
+#else
+            var dialogOperation = progressDialog.ShowDialog();
+            Action cancel = progressDialog.Hide;
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: Do you need to do anything with the result of `ShowDialog`?

File: ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs (hunk 11 above)
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevSupportManager.cs]: Call it something more accurate than AsyncInfo (like Cancel)?

---
In reply to: [83941933](https://github.com/ReactWindows/react-native-windows/pull/789#discussion_r83941933) [](ancestors = 83941933)

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs (hunk 16 above)
Comment (kevinvangelder) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: Probably, it was in the example I stole this from and they were doing a Console.write which didn't seem useful but I couldn't think of anything better to use so I figured I'd leave it blank to see if you had any suggestions.

File: ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs [hunk 18]
```diff
@@ -130,14 +137,37 @@ private T GetSetting<T>(string key, T defaultValue)
                     return (T)data;
                 }
             }
+#else
+            if (typeof(T) == typeof(bool))
+            {
+                Boolean result = Boolean.TryParse(ConfigurationManager.AppSettings[key], out result);
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/DevSupport/DevInternalSettings.cs]: This isn't doing what you think it's doing...
